elfpy/data/db_schema.py

```diff
@@ -0,0 +1,154 @@
+    lpTotalSupply: Mapped[float | None] = mapped_column(Numeric, default=None)
+    sharePrice: Mapped[float | None] = mapped_column(Numeric, default=None)
+    longsOutstanding: Mapped[float | None] = mapped_column(Numeric, default=None)
+    longAverageMaturityTime: Mapped[float | None] = mapped_column(Numeric, default=None)
+    shortsOutstanding: Mapped[float | None] = mapped_column(Numeric, default=None)
+    shortAverageMaturityTime: Mapped[float | None] = mapped_column(Numeric, default=None)
+    shortBaseVolume: Mapped[float | None] = mapped_column(Numeric, default=None)
+    withdrawalSharesReadyToWithdraw: Mapped[float | None] = mapped_column(Numeric, default=None)
+    withdrawalSharesProceeds: Mapped[float | None] = mapped_column(Numeric, default=None)
+
+
+class Transaction(Base):
+    """
+    Table/dataclass schema for Transactions
+    Mapped class that is a data class on the python side, and an declarative base on the sql side.
+    """
+
+    __tablename__ = "transactions"
+
+    # Default table primary key
+    # Note that we use postgres in production and sqlite in testing, but sqlite has issues with
+    # autoincrement with BigIntegers. Hence, we use the Integer variant when using sqlite in tests
+    id: Mapped[int] = mapped_column(
+        BigInteger().with_variant(Integer, "sqlite"), primary_key=True, init=False, autoincrement=True
+    )
+
+    #### Fields from base transactions ####
+    blockNumber: Mapped[int] = mapped_column(BigInteger, ForeignKey("poolinfo.blockNumber"), index=True)
+    transactionIndex: Mapped[int | None] = mapped_column(Integer, default=None)
+    nonce: Mapped[int | None] = mapped_column(Integer, default=None)
+    transactionHash: Mapped[str | None] = mapped_column(String, default=None)
+    # Transaction receipt to/from
+    # Almost always from wallet address to smart contract address
+    txn_to: Mapped[str | None] = mapped_column(String, default=None)
+    txn_from: Mapped[str | None] = mapped_column(String, default=None)
+    gasUsed: Mapped[int | None] = mapped_column(Numeric, default=None)
+
+    #### Fields from solidity function calls ####
+    # These fields map solidity function calls and their corresponding arguments
+    # The params list is exhaustive against all possible methods
+    input_method: Mapped[str | None] = mapped_column(String, default=None)
+
+    # Method: initialize
+    input_params_contribution: Mapped[float | None] = mapped_column(Numeric, default=None)
+    input_params_apr: Mapped[float | None] = mapped_column(Numeric, default=None)
+    input_params_destination: Mapped[str | None] = mapped_column(String, default=None)
+    input_params_asUnderlying: Mapped[bool | None] = mapped_column(Boolean, default=None)
+
+    # Method: openLong
+    input_params_baseAmount: Mapped[float | None] = mapped_column(Numeric, default=None)
+    input_params_minOutput: Mapped[float | None] = mapped_column(Numeric, default=None)
+    # input_params_destination
```

Comment:
What's the intent with all of these commented variables?

Reply:
Was going through each function call to solidity, and all of the parameters for each call. Some of the parameters are duplicates, but still wanted to keep track of parameters per function call, hence the commented variables